libsolidity/analysis/TypeChecker.cpp

```diff
@@ -920,6 +913,10 @@ bool TypeChecker::visit(Conditional const& _conditional)
 	}
 
 	_conditional.annotation().type = commonType;
+	_conditional.annotation().isPure =
+		_conditional.condition().annotation().isPure &&
```

Comment:
Ok, I cannot think of an actual use-case here, but the idea is that you want to have code that is compiled but inactive. I think we can add the warning for now: https://github.com/ethereum/solidity/issues/1761